modules/ROOT/pages/design-helpful-features-text-editor.adoc

```diff
@@ -82,3 +90,84 @@ types:
           application/json:
             type: person  <3>
 ----
+
+== Find Locations Where a Definition is Referenced
+
+You can navigate to locations where a given definition is referenced by right-clicking on the definition and selecting *Go to References*.
+
+For example, the type `person` is declared on line 5 at the first callout in the following example API specification. That type is subsequently used on line 14 at the second callout, and on line 22 at the third callout. If you right-click `person` on line 5 and select *Go to References*, a list of the `person` locations (callouts two and three) appears. Double clicking on a listed location will navigate you to the relevant line.
+
+----
+#%RAML 1.0
+title: Test References
+
+uses:
+  lib: library.raml 
+
+types:
+  person:  <1>
+    description: this is person's description
+    properties:
+      firstname: string
+  manager:
+    type: lib.manager 
+  Alertable: manager | person  <2>
+
+/person:
+  get:
+    responses:
+      200:
+        body:
+          application/json:
+            type: person[]  <3>
+----
+
+The action also supports the following:
+
+* RAML resource types and traits
+* RAML type expressions
+* Library aliases
+* References linked in JSON path navigation
+* References to nodes that apply “macros”, such as annotationTypes, securitySchemes, resourceTypes and traits
+
+== Select Code Ranges
+
+“Selection Ranges” allow you to select and to expand the selection of code blocks. 
+
+To select a range, click on an element and press Control + Shift + [Left Arrow | Right Arrow]. The Right Arrow key increases the range. The Left Arrow key decreases the range.
+
+For example, clicking on `body` in the following code, and pressing Control + Shift + Right Arrow, creates a selection range that only includes `body:`:
+
+image:apid-helpful-text-editor-1.png[30%,30%]
+
+Pressing Control + Shift + Right Arrow again results in the following selection range:
+
+image:apid-helpful-text-editor-2.png[30%,30%]
+
+Pressing Control + Shift + Left Arrow collapses the selection range, leaving only `body:` selected again.
+
+== Navigate to a Symbol
+
+You can right-click anywhere in the document and select *Go to Symbol*, which will present the list of defined symbols: 
+
+image:apid-helpful-text-editor-3.png[75%,75%]
+
+Selecting a symbol will navigate you to its location in the document.
+
+== Extract an Element to a Declaration
+
+You can take advantage of refactoring helpers to ease the continuous improvement of the design. One such helper is the ability to extract an inline declarable element to a top-level declaration, with the result being reusabiity.
+
+When you select a range that matches an inline declarable element, an icon appears to the left of the element:
+
+image:apid-helpful-text-editor-4.png[37%,37%]
+
+Tap the icon and select *Extract to Declaration*, which results in the following two changes to the document:
```

Comment:
We typically use "click" instead of tap for icons; is there a particular reason for "tap" here?

Reply:
Fixed